libs/vkd3d/state.c

```diff
@@ -616,18 +633,23 @@ static HRESULT d3d12_root_signature_init_root_descriptors(struct d3d12_root_sign
         const VkPushConstantRange *push_constant_range, struct vkd3d_descriptor_set_context *context,
         VkDescriptorSetLayout *vk_set_layout)
 {
-    VkDescriptorSetLayoutBinding *vk_binding_info, *vk_binding;
+    VkDescriptorSetLayoutBinding *vk_binding, *vk_binding_info = NULL;
     struct vkd3d_shader_resource_binding *binding;
     VkDescriptorSetLayoutCreateFlags vk_flags;
     struct d3d12_root_parameter *param;
     unsigned int i, j;
-    HRESULT hr;
+    HRESULT hr = S_OK;
+    bool raw_va;
 
-    if (!info->push_descriptor_count && !(root_signature->flags & VKD3D_ROOT_SIGNATURE_USE_INLINE_UNIFORM_BLOCK))
-        return S_OK;
+    raw_va = !!(root_signature->device->bindless_state.flags & VKD3D_RAW_VA_ROOT_DESCRIPTOR);
 
-    if (!(vk_binding_info = vkd3d_malloc(sizeof(*vk_binding_info) * (info->push_descriptor_count + 1))))
-        return E_OUTOFMEMORY;
+    if (info->push_descriptor_count || (root_signature->flags & VKD3D_ROOT_SIGNATURE_USE_INLINE_UNIFORM_BLOCK))
+    {
+        if (!(vk_binding_info = vkd3d_malloc(sizeof(*vk_binding_info) * (info->push_descriptor_count + 1))))
+            return E_OUTOFMEMORY;
+    }
+    else if (!raw_va)
+        return S_OK;
```

Comment:
Is vk_set_layout initialized if we actually end up not creating a descriptor set layout here?

Reply:
We zero-initialize the entire root signature struct on creation.